core/src/main/scala/org/apache/spark/util/io/ChunkedByteBufferOutputStream.scala

```diff
@@ -30,9 +31,14 @@ import org.apache.spark.storage.StorageUtils
  * @param chunkSize size of each chunk, in bytes.
  */
 private[spark] class ChunkedByteBufferOutputStream(
-    chunkSize: Int,
+    var chunkSize: Int,
     allocator: Int => ByteBuffer)
-  extends OutputStream {
+  extends OutputStream with Logging{
+
+  if (chunkSize < 0) {
+    logWarning(s"chunkSize should not be an negative value, replaced as 4MB default.")
+    chunkSize = 4 * 1024 * 1024
+  }
 
```

Comment:
As `chunkSize` is passed from many code path, and there is underlying integer overflow when convert from `Long` to `Int`. As we do not have a better solution, introducing a protection check may be a tradeoff way in the first step.

@JoshRosen @srowen